From: =?utf-8?q?Jos=C3=A9_Juli=C3=A1n_Espina?= <julian.espina@canonical.com>
Date: Wed, 3 Jun 2026 17:39:24 -0600
Subject: Fix execute permissions on lsvcgss service

Forwarded: no
---
 lustre/scripts/systemd/Makefile.am | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/lustre/scripts/systemd/Makefile.am b/lustre/scripts/systemd/Makefile.am
index 18700c6..107a6ec 100644
--- a/lustre/scripts/systemd/Makefile.am
+++ b/lustre/scripts/systemd/Makefile.am
@@ -2,8 +2,7 @@
 genunitfiles = lnet.service
 
 if HAVE_SYSTEMD
-systemdsystemunit_DATA = lnet.service
-systemdsystemunit_SCRIPTS = lsvcgss.service
+systemdsystemunit_DATA = lnet.service lsvcgss.service
 endif
 
 EXTRA_DIST = $(addsuffix .in,$(genunitfiles)) lsvcgss.service
